openedx/features/specializations/helpers.py

```diff
@@ -1,7 +1,147 @@
+from lms.djangoapps.grades.course_grade_factory import CourseGradeFactory
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.timed_notification.core import get_course_first_chapter_link
+from student.helpers import cert_info
+from student.models import CourseEnrollment
+from xmodule.modulestore.django import modulestore
 
 DISCOVERY_DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
 
 
+def date_time_from_now(delta_days):
+    """
+    Get delta date from now in string format
+    Args:
+        delta_days (int): No of days in past or future
+
+    Returns:
+        String date
+    """
+    return (datetime.now() + timedelta(days=delta_days)).strftime(DISCOVERY_DATE_FORMAT)
+
+
 def date_from_str(date_str, date_format=DISCOVERY_DATE_FORMAT):
     return datetime.strptime(date_str, date_format)
+
+
+def get_program_courses(user, specialization_uuid, detail=False):
+    """
+    Get course run in a program and data related to programs
+    Args:
+        user (object): User object
+        specialization_uuid (str): Program uuid
+        detail (boolean): True if want detailed info in course dict
+
+    Returns:
+        A tuple of courses and context from discovery programs
+    """
+    context = get_program_from_discovery(specialization_uuid)
+    courses = []
+
+    if not user.is_authenticated:
+        return context, courses
+
+    for course in [course for course in context.get('courses', []) if course.get('course_runs', [])]:
+        course_rerun = get_open_course_rerun(course['course_runs'])
+        course_id = CourseKey.from_string(course_rerun['key'])
+        course_rerun['course_id'] = course_id
+        course_rerun['enrolled'] = CourseEnrollment.is_enrolled(user, course_id)
+
+        if detail:
+            course_rerun['first_chapter_link'] = get_first_chapter_link(course_id)
+            course_rerun['completed'], course_rerun['in_progress'] = is_course_completed_or_in_progress(course_id, user)
+
+        courses.append(course_rerun)
+
+    context.update({'courses': courses})
+    return context, courses
+
+
+def get_open_course_rerun(course_runs):
+    """
+    Sort list of course reruns and get first open rerun
+    Args:
+        course_runs (list): list of all reruns in a course
+
+    Returns:
+        Open course rerun
+    """
+    open_course_rerun_list = [
+        rerun for rerun in course_runs
+        if rerun['enrollment_start'] and rerun['enrollment_end'] and date_from_str(
+            rerun['enrollment_start']) <= datetime.now() <= date_from_str(rerun['enrollment_end'])
+    ]
+
+    opened = bool(open_course_rerun_list)
+    course_rerun = (
+        sorted(
+            open_course_rerun_list, key=lambda open_rerun: date_from_str(open_rerun['enrollment_start'])
+        )[0] if opened else course_runs[0]
+    )
+
+    course_rerun['opened'] = opened
+    return course_rerun
+
+
+def get_first_chapter_link(course_id):
```

Comment:
We have a function to get link of any chapter of provided course... Pls have a look at `from openedx.features.philu_courseware.helpers import get_nth_chapter_link` if that could server our purpose and we don't have to write an additional function here.

Reply:
As per our discussion leaving this as is.